at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -107,42 +111,60 @@ class ScanVerbHandler extends AbstractVerbHandler {
-  /// @param - List<String>: List of keys from the secondary persistent store
-  /// @return - String: Returns the keys of current atsign
-  String _getLocalKeys(
-      InboundConnectionMetadata atConnectionMetadata, List<String?> keys) {
-    var keyString;
+  ///
+  /// **Parameters**
+  ///
+  /// [atConnectionMetadata] Metadata of the inbound connection.
+  ///
+  /// [keys] List of keys from the secondary persistent store.
+  ///
+  /// **Returns**
+  ///
+  /// Returns the list of keys of current atsign.
+  List<String> _getLocalKeys(
+      InboundConnectionMetadata atConnectionMetadata, List<String> keys) {
+    List<String> keysList = [];
     // Verify if the current user is authenticated or not
     // If authenticated get all the keys except for private keys
     // If not, get only public keys
     if (atConnectionMetadata.isAuthenticated) {
       //display all keys except private
-      keys.removeWhere((key) =>
-          key.toString().startsWith('privatekey:') ||
-          key.toString().startsWith('public:_') ||
-          key.toString().startsWith('private:'));
-      keyString = keys.toString();
+      keys.removeWhere((key) => _isPrivateKeyForAtSign(key));
+      keysList = keys;
     } else {
-      //When pol is performed, display keys that are private to the atsign.
+      // When pol is performed, display keys that are private to the atsign.
       if (atConnectionMetadata.isPolAuthenticated) {
-        keys.removeWhere((test) =>
-            (test
-                    .toString()
-                    .startsWith('${atConnectionMetadata.fromAtSign}:') ==
-                false) ||
-            test.toString().startsWith('public:_'));
-        keyString = keys
-            .toString()
-            .replaceAll('${atConnectionMetadata.fromAtSign}:', '');
+        keys.removeWhere((key) => _isPrivateKeyForAtSign(key,
+            fromAtSign: atConnectionMetadata.fromAtSign));
+        // Remove the atSigns from the inbound connection
+        // keys and add the modified key to the list.
+        // @murali:phone@sitaram => phone@sitaram
+        for (var key in keys) {
+          var modifiedKey =
+              key.replaceAll('${atConnectionMetadata.fromAtSign}:', '');
+          keysList.add(modifiedKey);
+        }
       } else {
         // When pol is not performed, display only public keys
-        keys.removeWhere((test) =>
-            test.toString().startsWith('public:_') ||
-            !test.toString().startsWith('public:'));
-        keyString = keys.toString().replaceAll('public:', '');
+        keys.removeWhere((key) => _isPrivateKeyForAtSign(key));
+        for (var key in keys) {
+          var modifiedKey = key.toString().replaceAll('public:', '');
+          keysList.add(modifiedKey);
+        }
       }
     }
-    return keyString;
+    return keysList;
+  }
+
+  /// Checks if a key starts with `public:`, `public:_`, `private:`, `privatekey:` given.
```

Comment:
keys starting with public: are public keys..not private keys

Reply:
This function wasn't working as expected. Need more knowledge and idea about it. Till that, I'm reverting this change.